components/script_plugins/unrooted_must_root.rs

```diff
@@ -38,6 +40,7 @@ impl UnrootedPass {
     }
 }
 
+// left to let the code compile
 /// Checks if a type is unrooted or contains any owned unrooted types
 fn is_unrooted_ty(cx: &LateContext, ty: &ty::TyS, in_new_function: bool) -> bool {
```

Comment:
I'm confused. It seems bad to have two copies of this. =)

Is there a reason to do so just because we didn't refactor everything yet?

Reply:
Yes, I haven't refactored the code yet. That's why I asked questions about that num "either def_id or generics".